commons-mfa/src/main/java/org/exoplatform/mfa/api/MfaService.java

```diff
@@ -81,4 +83,25 @@ public boolean hasRevocationRequest(String username, String mfaType) {
   public void deleteRevocationRequest(String username, String type) {
     mfaStorage.deleteRevocationRequest(username,type);
   }
+
+  public List<RevocationRequest> getAllRevocationRequests() {
+    return mfaStorage.findAll();
+  }
+  public RevocationRequest getRevocationRequestById(Long id) {
+    return mfaStorage.findById(id);
+  }
+
+  public void confirmRevocationRequest(Long id) {
+    RevocationRequest revocationRequest = mfaStorage.findById(id);
+    if (revocationRequest!=null && revocationRequest.getType().equals("otp")) {
+      //todo : change this when services are added in map by configuration
+      OtpService otpService = CommonsUtils.getService(OtpService.class);
```

Comment:
is it possible to add the dependency in the constructor instead ?

Reply:
Not this one. I have another task pending, which will allow me to remove this getService
So it will be done before the end of dev